db/src/urls.rs

```diff
@@ -69,19 +73,29 @@ async fn download_raw_html(url: &str) -> color_eyre::Result<String> {
     Ok(html)
 }
 
+fn clean_raw_html(raw_html: &str, url: &Url) -> color_eyre::Result<String> {
+    let mut raw_html_cursor = std::io::Cursor::new(raw_html);
+    let article = extractor::extract(&mut raw_html_cursor, url)
+        .map_err(|e| color_eyre::eyre::eyre!(e.to_string()))?;
+    Ok(article.content)
+}
+
```

Comment:
Used `readability::extractor` to extract the HTML content from the URl

Reply:
It's interesting that this extract takes the URL in, but on second thought it makes sense!

It's probably used at some point to (at a miniumum) convert relative links to absolute ones, or things like that.
I'm actually running into a similar (but reverse problem kinda) where in my blog I'm going from Markdown to HTML and currently when I link to an image its usually with a relative link like `../images/something.png` and when the browser sees that it knows to use the current url as the 'base'
My problem was that I wanted to add a CDN to the middle, and the CDN needs the full url to the image. And of course in my markdown -> html code path I don't currently have the url! So my next refactor there is going to be piping the URL down farther lol

Just rambling! No changes here, looks great!

Reply:
Your next refactor on coreyja.com?

I can take a look if you want me to?